draco/analysis/delay.py

```diff
@@ -416,52 +403,100 @@ def process(self, ss):
 
         if self.complex_timedomain:
             self.nfreq = len(ss.freq)
-            channel_ind = np.arange(self.nfreq)
-            ndelay = self.nfreq
+            self.channel_ind = np.arange(self.nfreq)
+            self.ndelay = self.nfreq
         else:
-            channel_ind = (
+            self.channel_ind = (
                 np.abs(ss.freq[:] - self.freq_zero) / self.freq_spacing
             ).astype(np.int64)
             if self.nfreq is None:
-                self.nfreq = channel_ind[-1] + 1
+                self.nfreq = self.channel_ind[-1] + 1
 
                 if self.skip_nyquist:
                     self.nfreq += 1
 
             # Assume each transformed frame was an even number of samples long
-            ndelay = 2 * (self.nfreq - 1)
+            self.ndelay = 2 * (self.nfreq - 1)
 
-        # Compute delays corresponding to output delay power spectrum
-        delays = np.fft.fftshift(np.fft.fftfreq(ndelay, d=self.freq_spacing))  # in us
+        # Compute delays corresponding to output delay power spectrum (in us)
+        self.delays = np.fft.fftshift(np.fft.fftfreq(self.ndelay, d=self.freq_spacing))
 
-        # Initialise the spectrum container
-        delay_spec = containers.DelaySpectrum(
-            baseline=baselines, delay=delays, attrs_from=ss
-        )
-        delay_spec.redistribute("baseline")
-        delay_spec.spectrum[:] = 0.0
+        # Get views of data and weights appropriate for the type of processing we're
+        # doing. Also get container for output (DelayTransform or DelaySpectrum)
+        data_view, weight_view, out_cont = self._process_data(ss)
+
+        # Evaluate frequency->delay transform
+        output_spectrum = self._evaluate(data_view, weight_view, out_cont)
```

Comment:
This is very minor. I think it could be more clear here that `output_spectrum` and `out_cont` are actually the same object. Possibly having `_evaluate` not return anything, although I'm not sure if that would actually be more clear.

Reply:
I've changed everything to `out_cont` and added a few clarifying comments - let me know if this doesn't help enough